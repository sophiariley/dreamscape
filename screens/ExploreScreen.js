// The screen the user interacts with in order to search Dreamscape.

import React, { useState} from "react";
import { StyleSheet, TextInput, View, KeyboardAvoidingView, Text } from "react-native";
import NavigationBar from "../components/navigationBar";
import ProfileSearchReturn from "../components/profileSearchReturn";
import {Foundation} from 'react-native-vector-icons';
import { collection, query, where, onSnapshot, getDocs, setDoc, doc } from "firebase/firestore";
import { db } from "../firebase-config";

const ExploreScreen = ({route}) => {
    const [search, setSearch] = useState('');
    const [match, setMatch] = useState('');
    const [matchFound, setMatchFound] = useState(false);
    const [matchID, setMatchID] = useState('');

    async function findUsernameMatch(username) {
        setMatch('');
        setMatchFound(false);
        const q = query(collection(db, "users"), where("username", "==", username));
        const querySnapshot = await getDocs(q);
        querySnapshot.forEach((doc) => {
            console.log(doc.data().username); //TODO - Alter this to help with UI display
            setMatch(username);
            setMatchFound(true);
            setMatchID(doc.id);
        });
    }

    function doBoth(text) {
        setSearch(text);
        findUsernameMatch(text);
    }

    //User Info
    const username = route.params.username;
    const userID = route.params.userID;


    const searchReturn = matchFound && match!='' && matchID!=''? <ProfileSearchReturn toSearchReturnUsername={match} toSearchReturnUserID={matchID} toNavBarUsername={username} toNavBarUserID={userID}/> : null;
    
    return (
        <View style={styles.container}>
            <View style={styles.header}>

            {/* The searchbar, which the user can interact with in order to search the app */}
            <KeyboardAvoidingView style={styles.searchBar}
            behavior="padding">
                <Foundation name='magnifying-glass' size={25} color='#3A6496' style={styles.icon}/>
                <TextInput
                    style={styles.searchBarText}
                    placeholder="Search"
                    value={search}
                    onChangeText={text => doBoth(text)}
                />
            </KeyboardAvoidingView>
            </View>

            {/* The navigation bar, which the user can use to navigate the app */}
            <KeyboardAvoidingView style={styles.footer}>
                <NavigationBar toNavBarUsername={username} toNavBarUserID={userID}/>
            </KeyboardAvoidingView>
            {searchReturn}
        </View>
    )
}

export default ExploreScreen;

// The styles used for this screen
const styles = StyleSheet.create({
    // A general container for components
    container: {
        backgroundColor: 'white',
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center'
    },
    // the styling of the header, used for the search bar
    header: {
        position: 'absolute',
        top: 300,
        alignItems:'center',
        width: '100%'
    },
    // the styling of the footer, used for the navigation bar
    footer: {
        position: 'absolute',
        bottom: 0,
        alignItems:'center'
    },

    // A container for the search bar
    searchBar: {
        backgroundColor: '#DADADA',
        opacity: .75,
        width: '95%',
        borderRadius: 5,
        top: -280,
        flexDirection: 'row'
    },

    // The styling for the text in the search bar
    searchBarText: {
        fontSize: 18,
        color: '#3A6496',
        marginLeft: 10,
        marginTop: 5,
        marginBottom: 5
    },

    // The styling for the icon in the search bar
    icon: {
        marginLeft: 10,
        marginTop: 5,
        marginBottom: 5
    }

})