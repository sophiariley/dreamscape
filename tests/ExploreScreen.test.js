import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react-native'; 
import ExploreScreen from './ExploreScreen';
import {create} from 'react-test-renderer';

// basic rendering test
test('renders correctly across screens', () => {
  const tree = create(<ExploreScreen />);
  expect(tree).toMatchSnapshot();
});


// Testing search box

const SEARCH_TEXT = 'search';

describe('Testing search input', () => {
  it('should find the search bar via testId', () => {
  const testIdName = 'searchBar';
  const {getByTestId} = render(<ExploreScreen />);
  const foundButton = getByTestId(testIdName);
  expect(foundButton).toBeTruthy();
  });

    it('Search bar is pressable', () => {
      const page = render(<ExploreScreen/>);
      const searchBar = page.getByTestId('searchBar');
      fireEvent.press(searchBar);
    });

    test('search bar should allow for input', () => {
      render(<ExploreScreen/>);
      fireEvent.changeText(screen.getByPlaceholderText('Search'), SEARCH_TEXT)
    })
});