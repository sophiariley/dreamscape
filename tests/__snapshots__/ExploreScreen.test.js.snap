// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly across screens 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "position": "absolute",
        "top": 300,
        "width": "100%",
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        [
          {
            "backgroundColor": "#DADADA",
            "borderRadius": 5,
            "flexDirection": "row",
            "opacity": 0.75,
            "top": -280,
            "width": "95%",
          },
          {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <TextInput
        placeholder="Search"
        style={
          {
            "color": "#3A6496",
            "fontSize": 18,
            "marginBottom": 5,
            "marginLeft": 10,
            "marginTop": 5,
          }
        }
        testID="searchBar"
        value=""
      />
    </View>
  </View>
  <View
    onLayout={[Function]}
    style={
      {
        "alignItems": "center",
        "bottom": 0,
        "position": "absolute",
      }
    }
  >
    <View
      style={
        {
          "alignContent": "center",
          "backgroundColor": "#D9D9D9",
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "space-around",
          "width": "100%",
        }
      }
    >
      <View />
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "height": 50,
            "justifyContent": "center",
            "marginBottom": 5,
            "marginLeft": 15,
            "marginRight": 15,
            "marginTop": 5,
            "opacity": 1,
            "width": 50,
          }
        }
      />
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "height": 50,
            "justifyContent": "center",
            "marginBottom": 5,
            "marginLeft": 15,
            "marginRight": 15,
            "marginTop": 5,
            "opacity": 1,
            "width": 50,
          }
        }
      />
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "height": 50,
            "justifyContent": "center",
            "marginBottom": 5,
            "marginLeft": 15,
            "marginRight": 15,
            "marginTop": 5,
            "opacity": 1,
            "width": 50,
          }
        }
      />
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "height": 50,
            "justifyContent": "center",
            "marginBottom": 5,
            "marginLeft": 15,
            "marginRight": 15,
            "marginTop": 5,
            "opacity": 1,
            "width": 50,
          }
        }
      />
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "height": 50,
            "justifyContent": "center",
            "marginBottom": 5,
            "marginLeft": 15,
            "marginRight": 15,
            "marginTop": 5,
            "opacity": 1,
            "width": 50,
          }
        }
      />
      <Text />
      <View />
    </View>
  </View>
</View>
`;
